src/cluster_legacy.c

```diff
@@ -885,37 +940,66 @@ void clusterUpdateMyselfIp(void) {
 }
 
 /* Update the hostname for the specified node with the provided C string. */
-static void updateAnnouncedHostname(clusterNode *node, char *new) {
+static void updateAnnouncedHostname(clusterNode *node, char *value) {
     /* Previous and new hostname are the same, no need to update. */
-    if (new && !strcmp(new, node->hostname)) {
+    if (value != NULL && !strcmp(value, node->hostname)) {
         return;
-    } else if (!new && (sdslen(node->hostname) == 0)) {
+    } else if (value == NULL && sdslen(node->hostname) == 0) {
         return;
     }
 
-    if (new) {
-        node->hostname = sdscpy(node->hostname, new);
-    } else if (sdslen(node->hostname) != 0) {
+    if (value != NULL) {
+        node->hostname = sdscpy(node->hostname, value);
+    } else {
         sdsclear(node->hostname);
     }
     clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG);
 }
 
-static void updateAnnouncedHumanNodename(clusterNode *node, char *new) {
-    if (new && !strcmp(new, node->human_nodename)) {
+static void updateAnnouncedHumanNodename(clusterNode *node, char *value) {
+    if (value != NULL && !strcmp(value, node->human_nodename)) {
         return;
-    } else if (!new && (sdslen(node->human_nodename) == 0)) {
+    } else if (value == NULL && sdslen(node->human_nodename) == 0) {
         return;
     }
 
-    if (new) {
-        node->human_nodename = sdscpy(node->human_nodename, new);
-    } else if (sdslen(node->human_nodename) != 0) {
+    if (value != NULL) {
+        node->human_nodename = sdscpy(node->human_nodename, value);
+    } else {
         sdsclear(node->human_nodename);
     }
     clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG);
 }
 
+static void updateAnnouncedClientIpV4(clusterNode *node, char *value) {
```

Comment:
There are 4 copies of boilerplate code now with client_ipv4 and client_ipv6. Do you think we could consider something like below? I think aux fields could use the same pattern, too.

```c

static void updateSdsExtensionField(char **field,  sds value) {
    if (value != NULL && !strcmp(value, *field)) {
        return;
    } else if (value == NULL && sdslen(*field) == 0) {
        return;
    }

    if (value != NULL) {
        *field = sdscpy(*field, value);
    } else {
        sdsclear(*field);
    }
    clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG);
}

static void updateAnnouncedClientIpV4(clusterNode *node, sds value) {
    updateExtensionFieldSds(&node->announce_client_ipv4, value);
}

static void updateAnnouncedClientIpV6(clusterNode *node, sds value) {
    updateExtensionFieldSds(&node->announce_client_ipv6, value);
}

static void updateHostname(clusterNode *node, sds value) {
    updateExtensionFieldSds(&node->hostname, value);
}

static void updateHumanNodename(clusterNode *node, sds value) {
    updateExtensionFieldSds(&node->human_nodename, value);
}

```

Reply:
Done. PTAL.